src/nerpa_pipeline/handle_rban.py

```diff
@@ -49,20 +57,57 @@ def build_nx_graph(rban_record, backbone_bonds, recognized_monomers, cut_lipids=
         # rBAN: N -> C
         s, e = bond['bond']['monomers']
 
-        if s in lipid_nodes or e in lipid_nodes:
+        if s in lipid_nodes or e in lipid_nodes:  # cut edges incident to lipid nodes
             continue
 
         bond_type = bond['bond']['bondTypes'][0]
-        if bond_type in backbone_bonds:
+        if bond_type in backbone_bonds:  # keeps only backbone edges
             # BGC: upstream_C -> N_downstream
+            '''ok, so in rBAN the edges are oriented in the opposite way. 
+            Does that hold for all types of bonds? I hope so'''
             G.add_edge(e, s, type=bond_type)
 
     return G
+def my_build_nx_graph(rban_record, recognized_monomers):
+    '''
+    The same a build_nx_graph, but does not cut any edges
+    and keeps in nodes 'smiles' and 'mass'
+    '''
+    monomeric_graph = rban_record['monomericGraph']['monomericGraph']
 
+    nodes = []
+    for monomer in monomeric_graph['monomers']:
+        idx = monomer['monomer']['index']
+        name =  monomer['monomer']['monomer']['monomer']
+        # TODO this is mass of the WHOLE aminoacid, the mass of monomer can differ
+        mass = Descriptors.ExactMolWt(MolFromSmiles(monomer['monomer']['monomer']['smiles']))
+        attr = {
+            'name': name.replace('C10:0-NH2(2)-Ep(9)-oxo(8)', 'Aeo'),
+            'isIdentified': name in recognized_monomers,
+            'mass': mass,
+            'smiles': monomer['monomer']['monomer']['smiles']
+            # 'isIdentified': monomer['monomer']['monomer']['isIdentified']
+        }
+        nodes.append((idx, attr))
+
+    G = nx.DiGraph()
+    G.add_nodes_from(nodes)
+    for bond in monomeric_graph['bonds']:
+        # rBAN: N -> C
+        s, e = bond['bond']['monomers']
+        bond_type = bond['bond']['bondTypes'][0]
+        G.add_edge(e, s, type=bond_type)
+
+    return G
 
 def hamiltonian_path(G, s, skip=None):
+    '''
+    This function just tries a single traversal of the graph, beginning at s.
+    If it does not succeed, it reports a failure
+    I feel that the same could be done with just a few lines of code...
+    '''
     node_to_id = {n:i for i, n in enumerate(G.nodes())}
-    if skip is None:
+    if skip is None:  # why not just set skip=[] be default? and why do we need skip in the first place?
```

Comment:
Replace the comment with the link to https://stackoverflow.com/questions/366422/how-can-i-avoid-issues-caused-by-pythons-early-bound-default-parameters-e-g-m